usr/src/cmd/zoneadmd/log.c

```diff
@@ -624,37 +669,36 @@ escape_json(const char *sbuf, int slen, char *dbuf, int dlen, int *scntp,
 			} else {
 				len = snprintf(append_buf, sizeof (append_buf),
 				    "\\u%04x", (int)(0xff & c));
-				assert(len < sizeof (append_buf));
+				VERIFY3S(len, <, sizeof (append_buf));
 			}
 			append = append_buf;
 			break;
 		}
 
-		len = strlcpy(&dbuf[i], append, dlen - i);
-		if (len >= dlen - i) {
+		len = strlen(append);
+
+		if (custr_append(dest, append) != 0) {
+			VERIFY3S(errno, ==, EOVERFLOW);
 			if (flushp != NULL) {
 				*flushp = B_TRUE;
 			}
 			break;
-		} else {
-			sbuf++;
-			i += len;
 		}
+
+		i += len;
```

Comment:
there's no need to track len any more: we can let custr_append() handle any overflow situation, as your code already does. Thus, we can drop dest_max as well.

Reply:
Done